src/conditions/check_fields.rs

```diff
@@ -555,24 +555,28 @@ pub struct CheckFields {
 }
 
 impl Conditional for CheckFields {
-    fn check(&self, e: &Event) -> bool {
-        self.predicates.iter().all(|(_, p)| p.check(e))
+    fn check(&self, e: Event) -> (bool, Event) {
+        let result = self.predicates.iter().all(|(_, p)| p.check(&e));
```

Comment:
Should we give the `check` method on the predicate an owned `Event` as well?

Reply:
I'd prefer to keep `&Event` where possible. VRL required an owned `Event`, but check field predicates don't use VRL.